### PR TITLE
NXP-19769: Add Folder Delta View

### PR DESCRIPTION
**Pull request by michaelva**

### CODE REVIEW

File: src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java [hunk 1]
```diff
@@ -37,12 +38,17 @@
 
     private static final URLTemplate SEARCH_IN_ROOT_URL = new URLTemplate("/drive/root/view.search");
 
+    private static final URLTemplate DELTA_IN_ROOT_URL = new URLTemplate("/drive/root/view.search");
```
Comment (kevinleturc) [inline src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java]: Should the URL be /drive/root/view.delta ?

File: src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java [hunk 2]
```diff
@@ -37,12 +38,17 @@
...
+
     private static final URLTemplate GET_FOLDER_URL = new URLTemplate("/drive/items/%s");
 
     private static final URLTemplate GET_CHILDREN_URL = new URLTemplate("/drive/items/%s/children");
 
     private static final URLTemplate SEARCH_IN_FOLDER_URL = new URLTemplate("/drive/items/%s/view.search");
 
+    private static final URLTemplate DELTA_IN_FOLDER_URL = new URLTemplate("/drive/items/%s/view.search");
```
Comment (kevinleturc) [inline src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java]: Should the URL be /drive/items/%s/view.delta ?

File: src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java [hunk 3]
```diff
@@ -104,6 +110,28 @@ public static OneDriveFolder getRoot(OneDriveAPI api) {
         return () -> new OneDriveItemIterator(getApi(), url);
     }
 
+    public OneDriveItemIterator delta() {
+        return delta(null);
+    }
+
+    public OneDriveItemIterator delta(String deltaLink) {
+        URL url;
+            if (deltaLink!=null) {
```
Comment (kevinleturc) [inline src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java]: I would rather not having this condition here, by leaving the code using deltaLink here, and the other part directly in delta().

File: src/main/java/org/nuxeo/onedrive/client/OneDriveItemIterator.java [hunk 4]
```diff
@@ -61,4 +61,8 @@ public boolean hasNext() throws OneDriveRuntimeException {
         return nextMetadata;
     }
 
+    public String getDeltaLink() {
```
Comment (kevinleturc) [inline src/main/java/org/nuxeo/onedrive/client/OneDriveItemIterator.java]: I would rather having an OneDriveItemDeltaIterator.
We could : 
- add a hook method in JsonObjectIterator, like onResponse(JsonObject json)
- add a protected constructor in OneDriveItemIterator as this OneDriveItemIterator(OneDriveAPI, JsonObjectIterator)
- add OneDriveItemDeltaIterator class extending OneDriveItemIterator and providing an anonymous implementation of JsonObjectIterator which implements onResponse and save the deltaLink.

File: src/main/java/org/nuxeo/onedrive/client/JsonObjectIterator.java [hunk 5]
```diff
@@ -79,6 +85,10 @@ private void loadNextPage() throws OneDriveRuntimeException {
             }
 
             JsonValue nextUrl = json.get("@odata.nextLink");
+
+            JsonValue delta = json.get("@odata.deltaLink");
+            deltaLink = delta.asString();
```
Comment (kevinleturc) [inline src/main/java/org/nuxeo/onedrive/client/JsonObjectIterator.java]: delta could be null

File: src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java (hunk 1 above)
Comment (michaelva) [inline src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java]: Fixed

File: src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java (hunk 2 above)
Comment (michaelva) [inline src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java]: Fixed

File: src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java (hunk 3 above)
Comment (michaelva) [inline src/main/java/org/nuxeo/onedrive/client/OneDriveFolder.java]: Made the change

File: src/main/java/org/nuxeo/onedrive/client/JsonObjectIterator.java (hunk 5 above)
Comment (michaelva) [inline src/main/java/org/nuxeo/onedrive/client/JsonObjectIterator.java]: Made the change to handle that

**Pull request merged**